services/wallet/router/router_v2_ut_test.go

```diff
@@ -0,0 +1,145 @@
+	w_common "github.com/status-im/status-go/services/wallet/common"
+
+	// mock_fees "github.com/status-im/status-go/services/wallet/router/mock/fees"
+	"github.com/status-im/status-go/services/wallet/router/pathprocessor"
+	"github.com/status-im/status-go/services/wallet/router/pathprocessor/mock_pathprocessor"
+	"github.com/status-im/status-go/services/wallet/token"
+	mock_token "github.com/status-im/status-go/services/wallet/token/mock/token"
+)
+
+func setupRouter(t *testing.T) (*Router, mock_token.MockManagerInterface, *MockCandidateResolver, *MockEstimator, *mock_pathprocessor.MockPathProcessor, *mock_pathprocessor.MockPathProcessor, *gomock.Controller) {
+	ctrl := gomock.NewController(t)
+
+	mockTokenManager := mock_token.NewMockManagerInterface(ctrl)
+	mockPathProcessor := mock_pathprocessor.NewMockPathProcessor(ctrl)
+	mockPathProcessor.EXPECT().Name().Return("mockPathProcessor").AnyTimes()
+	mockPathProcessorBridge := mock_pathprocessor.NewMockPathProcessor(ctrl)
+	mockPathProcessorBridge.EXPECT().Name().Return(pathprocessor.ProcessorBridgeHopName).AnyTimes()
+
+	candidateResolver := NewMockCandidateResolver(ctrl)
+	mockEstimator := NewMockEstimator(ctrl)
+	router := NewRouter(nil, nil, mockTokenManager, nil, nil, nil, nil, nil)
+	router.AddPathProcessor(mockPathProcessor)
+	router.AddPathProcessor(mockPathProcessorBridge)
+	router.CandidateResolver = candidateResolver
+	router.Estimator = mockEstimator
+	return router, *mockTokenManager, candidateResolver, mockEstimator, mockPathProcessor, mockPathProcessorBridge, ctrl
+}
+
+func TestRouterResolveCandidatesForNetwork(t *testing.T) {
+	router, mockTokenManager, mockCandidateResolver, _, _, _, ctrl := setupRouter(t)
+	defer ctrl.Finish()
+
+	input := &RouteInputParams{
+		TestnetMode:          false,
+		SendType:             Bridge,
+		AddrFrom:             common.HexToAddress("0x1"),
+		AddrTo:               common.HexToAddress("0x2"),
+		AmountIn:             (*hexutil.Big)(big.NewInt(testAmount1USDC)),
+		TokenID:              pathprocessor.UsdcSymbol,
+		DisabledFromChainIDs: []uint64{},
+		DisabledToChainIDs:   []uint64{},
+	}
+
+	// testParams: &routerTestParams{
+	token := &token.Token{
+		ChainID:  1,
+		Symbol:   pathprocessor.UsdcSymbol,
+		Decimals: 6,
+	}
+
+	networks := []*params.Network{
+		{
+			ChainID: w_common.OptimismMainnet,
+		},
+	}
+
+	network := &params.Network{
+		ChainID: token.ChainID,
+	}
+
+	// Test happy path for bridge hop
+	require.True(t, validateInput(input, network))
+	mockTokenManager.EXPECT().FindToken(network, pathprocessor.UsdcSymbol).Return(token).Times(1)
+	// TODO Define the expected return values - empty for 1st processor, non-empty for 2nd processor
+	mockCandidateResolver.EXPECT().resolveCandidatesForProcessor(gomock.Any(), input, network, token, gomock.Any(), big.NewInt(testAmount1USDC), false, router.pathProcessors[pathprocessor.ProcessorBridgeHopName], networks).Times(1).Return([]*PathV2{
```

Comment:
I am in general to avoid mocking for things we control, but here we are even mocking function of the same package/file.

IMO this is a red flag in our strategy

The more we mock that way the more the test is weak and we gonna have false positive:
test red but code is actually green, we should aim for test red when code is red